app.rb

```diff
@@ -0,0 +1,226 @@
+    @persons << stud
+    print "\n\nID: #{stud.id} Name: #{stud.name} Age: #{stud.age} Parent Permission: #{stud.parent_permission}"
+    print "\nNew Student is created successfully!\n"
+  end
+
+  def create_teacher
+    name = name_title('name')
+    age = age_entry
+    specialization = name_title('specialization')
+    teacher = Teacher.new(age, specialization, name)
+    @persons << teacher
+    print "\n\nID: #{teacher.id} Name: #{teacher.name} Age: #{teacher.age}"
+    print "\nNew Teacher is created successfully!\n"
+  end
+
+  def create_a_person
+    add_item = true
+    while add_item
+      if person_type == 1
+        create_student
+      else
+        create_teacher
+      end
+      add_item = choice_bool(%w[Y y], "Press [Y/y] to add another person\nOr", 's')
+    end
+  end
+
+  def create_a_book
+    add_item = true
+    while add_item
+      print "\nCreate a book\n"
+      title = name_title('title')
+      author = name_title('author')
+      book = Book.new(title, author)
+      @books << book
+      print "\n\nTitle: #{book.title} Author: #{book.author}"
+      print "\nNew Book is created successfully!\n"
+      add_item = choice_bool(%w[Y y], "Press [Y/y] to add another book\nOr", 's')
+      system('clear')
+    end
+  end
+
+  def check_books_persons
+    if @books.empty? || @persons.empty?
+      puts 'Book list Or Person list is empty!'
+      puts 'Please try entering some'
+      false
+    else
+      true
+    end
+  end
+
+  def sel_book_by_no
+    sel_mode = false
+    until sel_mode
+      print "\nSelect a book from the following list by number\n"
+      @books.map.with_index do |book, index|
+        puts "No: #{index + 1}) Title: '#{book.title}', Author: #{book.author}"
+      end
+      print "\nEnter your choice: "
+      sel_mode = (1..@books.length).include?(opt = gets.chomp.to_i)
+    end
+    opt - 1
+  end
+
+  def sel_person_by_no
+    sel_mode = false
+    until sel_mode
+      print "\nSelect a person from the following list by number\n"
+      @persons.map.with_index do |person, index|
+        puts "No: #{index + 1}) Name: '#{person.name}', Age: #{person.age}"
+      end
+      print "\nEnter your choice: "
+      sel_mode = (1..@persons.length).include?(opt = gets.chomp.to_i)
+    end
+    opt - 1
+  end
+
+  def create_a_rental
+    add_item = check_books_persons
+    while add_item
+      sel_book = @books[sel_book_by_no]
+      sel_person = @persons[sel_person_by_no]
+      print "\nEnter a date [format yyyy/mm/dd]: "
+      date = gets.chomp
+      @rentals << Rental.new(date, sel_person, sel_book)
+      print "\nDate: #{date} Book: #{sel_book.title} Name: #{sel_person.name}"
+      print "\nNew Rentals Added Successfully!\n"
+      add_item = choice_bool(%w[Y y], "Press [Y/y] to add another rental\nOr", 's')
+    end
+  end
+
+  # rubocop:disable Metrics/MethodLength
+  # rubocop:disable Metrics/PerceivedComplexity
+  def list_all_rentals # rubocop:disable Metrics/CyclomaticComplexity
```

Comment:
- [ ] As I mentioned above as per the general `requirements` you shouldn't disable the `robocop` linters in any situation. Because `linters` are meant to help you to write **code** more efficiently and cleanly. Please remove the `linter` disable comment and try to `resolve` the `linters` errors.